client/state/purchases/selectors/should-revert-atomic-site-before-deactivation.js

```diff
@@ -0,0 +1,37 @@
+import { isAtomicSupportedProduct } from '@automattic/calypso-products';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { getByPurchaseId } from './get-by-purchase-id';
+import { getSitePurchases } from './get-site-purchases';
+
+import 'calypso/state/purchases/init';
+
+/**
+ * Whether a purchase needs to trigger an Atomic revert prior its cancellation/removal.
+ *
+ * @param   {object} state       global state
+ * @param   {number} purchaseId  the purchase id
+ * @returns {boolean} True if the Atomic revert is needed, false otherwise.
+ */
+export const shouldRevertAtomicSiteBeforeDeactivation = ( state, purchaseId ) => {
+	if ( ! purchaseId ) {
+		return false;
+	}
+
+	const purchase = getByPurchaseId( state, purchaseId );
+
+	if ( ! purchase ) {
+		return false;
+	}
+
+	if ( ! isSiteAutomatedTransfer( state, purchase.siteId ) ) {
+		return false;
+	}
+
+	// Retrieves all Atomic supported purchases for the site.
+	// If there is at least one active subscription except the given one,
+	// the site needs to be kept in the Atomic infra.
+	return getSitePurchases( state, purchase.siteId ).some(
+		( sitePurchase ) =>
+			sitePurchase.siteId !== purchaseId && isAtomicSupportedProduct( purchase.productSlug )
```

Comment:
```
sitePurchase.**siteId** !== purchaseId
```
Is this intended? :)

Reply:
Good catch! Fixed in https://github.com/Automattic/wp-calypso/pull/55381/commits/7d2d9d70abbd1c729099b7d2c666f34f7473b227